modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismCaretInfo.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.sun.javafx.text;
+
+import javafx.geometry.Rectangle2D;
+import javafx.scene.text.CaretInfo;
+
+/**
+ * CaretInfo as reported by the PrismTextLayout.
+ */
+public final class PrismCaretInfo extends CaretInfo {
+
+    private final Rectangle2D[] parts;
+
+    public PrismCaretInfo(Rectangle2D[] parts) {
+        this.parts = parts;
+    }
+
+    @Override
+    public int getSegmentCount() {
+        return parts.length;
+    }
+
+    @Override
+    public Rectangle2D getSegmentAt(int index) {
+        return parts[index];
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("PrismCaretInfo{parts=[");
```

Comment:
I think you shouldn't leak implementation details via the `toString()` method (the name of the implementing class, as well as `parts=`, which is not a term that appears in the API). Maybe just drop `PrismCaretInfo{parts=` entirely.

Reply:
makes sense.

On a related note, we don't seem to have a consistent standard for toString() output (probably no one does!), so maybe it's worth polling the devs about it.

Standardized log output might be useful when one has to go through masses of old/transient logs, or when scraping the logs for analytics.  A JSON output might be useful since it offers a cheap structured form, although it suffers from two issues: verbosity and difficulty to encode binary data and object types easily.

Any ideas?